### PR TITLE
Include option to filter by course_id in the assignments endpoint

### PR DESCRIPTION
**Pull request by marcospri**
For:

- https://github.com/hypothesis/lms/issues/6393

### CODE REVIEW

File: lms/services/assignment.py [hunk 1]
```diff
@@ -238,6 +241,11 @@ def get_assignments(self, h_userid: str | None = None) -> Select[tuple[Assignmen
                 .where(User.h_userid == h_userid)
             )
 
+        if course_id:
+            assignments_query = assignments_query.join(AssignmentGrouping).where(
+                AssignmentGrouping.grouping_id == course_id
+            )
```
Comment (robertknight) [inline lms/services/assignment.py:247]: How does this work with assignments that use groups and/or sections?

File: lms/views/dashboard/api/assignment.py [hunk 2]
```diff
@@ -20,10 +21,10 @@
 
 
 class ListAssignmentsSchema(PaginationParametersMixin):
-    """Query parameters to fetch a list of assignments.
+    """Query parameters to fetch a list of assignments."""
 
-    Only the pagination related ones from the mixin.
-    """
+    course_id = fields.Integer(required=False, validate=validate.Range(min=1))
+    """Filter assignments that belong to the course with this ID."""
```
Comment (robertknight) [inline lms/views/dashboard/api/assignment.py:27]:
```suggestion
    """Return assignments that belong to the course with this ID."""
```

**Review by robertknight — COMMENTED** [review]
The code seems straightforward but made me realize that I'm lacking an understanding of how assignment groupings work and how a query like this would interact with the different kinds of grouping.

File: lms/services/assignment.py (hunk 1 above)
Comment (marcospri) [reply lms/services/assignment.py:247]: Assignments that use groupings will have rows in AssignmentGrouping for the course and each of the segments (either sections or groups).

But IMO while we are using the same table to represent both courses and segments, at the API level we should expose two different filters when we eventually expose segments in the dashboard.

**Review by robertknight — APPROVED** [review]

**Pull request merged**